Helper/IntegrationManagement.php

```diff
@@ -0,0 +1,468 @@
+                    if (isset($integrationData[Info::DATA_CONSUMER_ID])) {
+                        $this->oauthService->deleteConsumer($integrationData[Info::DATA_CONSUMER_ID]);
+                    }
+                    return true;
+                }
+            } else {
+                throw new Exception(
+                    __('The Bolt integration no longer exists.')
+                );
+            }
+        } catch (Exception $e) {
+            $this->bugsnag->notifyException($e);
+            return false;
+        }
+    }
+    
+    /**
+     * Create Magento integration for Bolt API.
+     *
+     * @return IntegrationModel|null
+     */
+    public function processMagentoIntegraion($storeId = null)
+    {
+        $boltIntegration = null;
+        try {
+            $integrationConfigs = $this->integrationConfig->getIntegrations();
+            if (isset($integrationConfigs[self::BOLT_INTEGRATION_NAME])) {
+                $integrationData = [
+                    IntegrationModel::NAME => self::BOLT_INTEGRATION_NAME,
+                    IntegrationModel::EMAIL => $integrationConfigs[self::BOLT_INTEGRATION_NAME][IntegrationConverter::KEY_EMAIL],
+                    IntegrationModel::ENDPOINT => $this->getEndpointUrl($storeId),
+                    IntegrationModel::IDENTITY_LINK_URL => $this->getIdentityLinkUrl($storeId),
+                    IntegrationModel::SETUP_TYPE => IntegrationModel::TYPE_CONFIG,
+                ];
+                $boltIntegration = $this->integrationService->findByName(self::BOLT_INTEGRATION_NAME);
+                if (!$boltIntegration->getId()) {
+                    $boltIntegration = $this->integrationService->create($integrationData);
+                } else {
+                    $integrationData[IntegrationModel::ID] = $boltIntegration->getId();
+                    $boltIntegration = $this->integrationService->update($integrationData);
+                }
+            }  
+        } catch (Exception $e) {
+            $this->bugsnag->notifyException($e);
+        } finally {
+            return $boltIntegration;
+        }
+    }
+    
+    /**
+     * Link access token of Magento integration to Bolt merchant account via OAuth handshake.
+     *
+     * @param int|string $storeId
+     *
+     * @return bool
+     */
+    public function linkAccessTokenBoltMerchantAccountByOAuth($storeId = null)
+    {
+        $boltIntegration = $this->processMagentoIntegraion();
+        if (!$boltIntegration || !$boltIntegration->getId()) {
+            return false;
+        }
+        try {
+            if ($this->getIntegrationStatus($storeId) == self::BOLT_INTEGRATION_STATUS_CREATED) {
+                // Re-send API keys to Bolt
+                // For security reason, we have to remove existing token associated with consumer and issue a new one
+                $this->oauthService->deleteIntegrationToken($boltIntegration->getConsumerId());
```

Comment:
Why do you want to delete token here?

Reply:
It is because for security reason, the oauth verifier token is one-time use for oauth handshake, and it becomes invalid when trying to resend integration tokesn. Then we have to delete current integration tokens to generate a new oauth_verifier.